### PR TITLE
Local engine build mode

### PR DESCRIPTION
**Pull request by dnfield**
Relands #51687 with a bad merge fixed up.

@zanderso @jmagman

### CODE REVIEW

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart [hunk 1]
```diff
@@ -209,9 +209,20 @@ List<String> _xcodeBuildSettingsLines({
...
+    String engineBuildMode = 'release';
+    if (engineOutPath.toLowerCase().contains('debug')) {
+      engineBuildMode = 'debug';
+    } else if (engineOutPath.toLowerCase().contains('profile')) {
+      engineBuildMode = 'profile';
+    }
     xcodeBuildSettings.add('FLUTTER_ENGINE=${globals.fs.path.dirname(globals.fs.path.dirname(engineOutPath))}');
-    xcodeBuildSettings.add('LOCAL_ENGINE=${globals.fs.path.basename(engineOutPath)}');
+    xcodeBuildSettings.add('LOCAL_ENGINE=$engineOutPath');
+    // Only write this or local engines, where it is supposed to be sticky to
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/ios/xcodeproj.dart]: or -> for

**Review by zanderso — APPROVED** [review]
lgtm w/ comment fix

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/lib/src/ios/xcodeproj.dart]: Done

**Pull request merged**